### ISSUE TITLE
no relation between trait_scores and trait_grades

### ISSUE THREAD

**Issue by nowakowski** [eSpectrum-IT/brassica#58]
On the CSDB diagram there is a link between `trait_scores` and `trait_grades`, however no actual FK field exists in `trait_scores` which would link to `trait_grades`. Requesting advice on how to proceed.

**Comment by wjurkowski** [comment]
I propose to merge and leave only trait_scores

2015-03-17 13:40 GMT+00:00 Piotr Nowakowski <notifications@github.com>:

**Comment by nowakowski** [comment]
OK, I think I understand the logic which went into creation of this table. Please correct me if I'm wrong: the express purpose of `trait_grades` is to provide some additional information regarding the interpretation of score values for a very specific trait descriptor, namely **hri_HRBB_01_GR**. As such there does seem to be a reason behind linking `trait_grades` to `trait_scores` although it obviously makes sense to retain `trait_grades` as an extension (of sorts) of `trait_descriptors`.

If you confirm that the above is correct then I will drop the link and close the issue.

**Comment by wjurkowski** [comment]
yes, but can we include the content of trait_grades in trait_scores? Is
there any reason to keep them separate?

2015-03-27 10:33 GMT+00:00 Piotr Nowakowski <notifications@github.com>:

**Comment by nowakowski** [comment]
@wjurkowski RDB design methodology suggests that it should remain a separate table. Since `trait_scores` references many different trait descriptors (38 in fact, if you exclude the dummy values) and only one specific trait descriptor has this sort of grade information attached, getting rid of `trait_grades` and moving all its fields to `trait_scores` would have two negative effects:

- **Denormalization**: we would need to replicate the value of `trait_grades.description` in every `trait_scores` record for which a given value is registered and the associated descriptor is *hri_HRBB_01_GR*. This leads to data duplication, which is storngly discouraged in RDBMS (sometimes we are forced to do it for performance reasons but this does not appear to be the case here).
- **Sparse records**. The `description` field would remain blank in every record for which the associated descriptor is not *hri_HRBB_01_GR*. This is questionable from a design efficiency standpoint.

Bottom line: we **can** get rid of `trait_grades` but I would recommend against it. If you feel the table should be dropped, say so and I will prepare the required migrations.

**Comment by wjurkowski** [comment]
Let's leave it as it is then.

2015-03-27 10:46 GMT+00:00 Piotr Nowakowski <notifications@github.com>:

**Comment by nowakowski** [comment]
Roger that.
